manifests/osfamily/aix.pp

```diff
@@ -44,10 +44,14 @@
       $aix_ver_number = '6.1'
     }
   }
-  if $::puppet_agent::source {
-    $source = $::puppet_agent::source
+  if $::puppet_agent::absolute_source {
+    $source = $::puppet_agent::absolute_source
+  } elsif $::puppet_agent::alternate_pe_source {
+    $source = "${::puppet_agent::alternate_pe_source}/packages/${pe_server_version}/aix-${aix_ver_number}-power/${::puppet_agent::package_name}-${::puppet_agent::package_version}-1.aix${aix_ver_number}.ppc.rpm"
+  } elsif $::puppet_agent::source {
+    $source = "${::puppet_agent::source}/packages/${pe_server_version}/aix-${aix_ver_number}-power/${::puppet_agent::package_name}-${::puppet_agent::package_version}-1.aix${aix_ver_number}.ppc.rpm"
   } else {
-    $source = "puppet:///pe_packages/${pe_server_version}/aix-${aix_ver_number}-power/${::puppet_agent::package_name}-${::puppet_agent::package_version}-1.aix${aix_ver_number}.ppc.rpm"
+    $source = "${::puppet_agent::aix_source}/${pe_server_version}/aix-${aix_ver_number}-power/${::puppet_agent::package_name}-${::puppet_agent::package_version}-1.aix${aix_ver_number}.ppc.rpm"
```

Comment:
How tough would it be to eliminate the measurable amount of duplication in the paths to the RPM? We appear to be changing the root, and the rest of the structure remains the same. Does Ruby take care of that for us?

This is a global question across all the constructs similar to this.

Reply:
We just talked in person about this: there's enough variation in the calculation of the parts of the path (i.e. the platform name in PE paths needs to be changed for AIX/Fedora/Amazon) that trying to centralize the data won't look any more maintainable than this.

Unfortunately I think we should keep it the way it looks so it's more understandable down the road.